### PR TITLE
Add uniformDiscrete distribution

### PR DESCRIPTION
**Pull request by jmh530**

### CODE REVIEW

**Comment by 9il** [comment]
Why is it needed?

**Comment by jmh530** [comment]
1. `mir.random` has a special overload of [`UniformVariable`](http://mir-random.libmir.org/mir_random_variable.html#UniformVariable) for the discrete case. I assumed it would be good to cover all the distributions in `mir.random` for consistency. 
2. The algorithms for pdf/cdf/etc are a little different in the continuous case vs. the discrete case. I figured it would be simpler to do it as a separate module, rather than an overload like that. For instance, if you have `uniformCDF(x, 1, 2)`, you shouldn't have to specify anything that you really intend that to be continuous. It reduces the chance of mixups by forcing the user to write `uniformDiscreteCDF(x, 1, 2)`. 
3. I'm just going through a list of commonly used statistical distributions and checking them off as I go. This was on it and figured it wouldn't take me very long to implement.

**Comment by jmh530** [comment]
@9il This distribution is also available in [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.randint.html#scipy.stats.randint) and [Matlab ](https://www.mathworks.com/help/stats/uniform-distribution-discrete.html)

File: source/mir/stat/distribution/uniform_discrete.d [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+See_also:
+    $(LINK2 https://en.wikipedia.org/wiki/Discrete_uniform_distribution, discrete uniform probability distribution)
++/
+@safe pure nothrow @nogc
+double uniformDiscretePMF(const size_t x, const size_t lower = 0, const size_t upper = 1)
+    in(x >= lower, "x must be greater than or equal to lower bound in discrete uniform probability distribution")
+    in(x <= upper, "x must be less than or equal to upper bound in discrete uniform probability distribution")
+    in(lower <= upper, "lower must be less than or equal to upper")
+{
+    return 1.0L / (upper - lower + 1);
```
Comment (9il) [inline source/mir/stat/distribution/uniform_discrete.d]: Why do we cast to `real`?

**Comment by 9il** [comment]
> @9il This distribution is also available in [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.randint.html#scipy.stats.randint) and [Matlab](https://www.mathworks.com/help/stats/uniform-distribution-discrete.html)

We can merge it, you are the main author of the package. I am not sure about the `real`. `80bit` FP is very slow on modern CPUs.

File: source/mir/stat/distribution/uniform_discrete.d (hunk 1 above)
Comment (jmh530) [reply source/mir/stat/distribution/uniform_discrete.d]: Fixed.

**Comment by jmh530** [comment]
@9il Thanks. Even if I'm the primary author, it is under the mir umbrella and I value your input.

**Pull request merged**